### PR TITLE
feat(flags): Add initial flags blueprint

### PR DESCRIPTION
**Pull request by cmanallen**
Related to:
* https://github.com/getsentry/sentry/issues/59428

### CODE REVIEW

File: src/sentry/flags/docs/api.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+**Attributes**
+
+| Column      | Type   | Description                                      |
+| ----------- | ------ | ------------------------------------------------ |
+| action      | string | Enum of `created` and `modified`.                |
+| environment | string | The environment the change applies to.           |
+| flag        | string | The name of the flag changed.                    |
+| modified_at | string | ISO-8601 timestamp of when the flag was changed. |
+| modified_by | string | The user responsible for the change.             |
```
Comment (michellewzhang) [inline src/sentry/flags/docs/api.md:37]: for `modified_by` should we specify if it's user id/email/name etc?

File: src/sentry/flags/docs/api.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+### Browse Flag Logs [GET]
+
+Retrieve a collection of flag logs.
+
+**Attributes**
+
+| Column      | Type   | Description                                      |
+| ----------- | ------ | ------------------------------------------------ |
+| action      | string | Enum of `created` and `modified`.                |
```
Comment (michellewzhang) [inline src/sentry/flags/docs/api.md:33]:
```suggestion
| action      | string | Enum of `created` or `modified`.                |
```

File: src/sentry/flags/docs/api.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  ```json
+  {
+    "data": {
+      "action": "modified",
+      "environment": "development",
+      "flag": "new-flag-name",
+      "modified_at": "2024-11-19T19:12:55",
+      "modified_by": "user@site.com",
+      "modified_by_type": "email"
```
Comment (bruno-garcia) [inline src/sentry/flags/docs/api.md:73]: I think the only way we can map users from a 3rd system to ours is through email. Should we set this to be only email and remove the `type` field? 

In other words: no username, name etc supported

Comment (cmanallen) [reply src/sentry/flags/docs/api.md:73]: It varies by provider what they return on the webhook.  We can't make assumptions about the type of the field (similar to the sdk's set_user method).

File: src/sentry/flags/docs/api.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+Retrieve a single flag log instance.
+
+- Response 200
+
+  ```json
+  {
+    "data": {
+      "action": "modified",
+      "environment": "development",
+      "flag": "new-flag-name",
```
Comment (bruno-garcia) [inline src/sentry/flags/docs/api.md:70]: Could we know the value too? 

If we knew `before` and `after` we could help debug by showing what it changed to

File: src/sentry/flags/docs/api.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  ```json
+  {
+    "data": [
+      {
+        "action": "created",
+        "environment": "production",
+        "flag": "my-flag-name",
+        "modified_at": "2024-01-01T05:12:33",
+        "modified_by": "2552",
+        "modified_by_type": "id"
```
Comment (michellewzhang) [inline src/sentry/flags/docs/api.md:51]: would it be helpful to know the starting state/value of the flag?

File: src/sentry/flags/docs/api.md (hunk 4 above)
Comment (cmanallen) [reply src/sentry/flags/docs/api.md:70]: Varies by provider.  I can add an optional field but this is not a trivial line of work.  I'd like to start with the minimum -- _something changed_ -- which the user will understand implicitly or go to their service provider to understand.

File: src/sentry/flags/docs/api.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+**Attributes**
+
+| Column           | Type   | Description                                          |
+| ---------------- | ------ | ---------------------------------------------------- |
+| action           | string | Enum of `created` or `modified`.                     |
+| flag             | string | The name of the flag changed.                        |
+| modified_at      | string | ISO-8601 timestamp of when the flag was changed.     |
+| modified_by      | string | The user responsible for the change.                 |
+| modified_by_type | string | Enum of `email`, `id`, or `name`.                    |
+| tags             | string | A collection of provider-specified scoping metadata. |
```
Comment (michellewzhang) [inline src/sentry/flags/docs/api.md:37]: should the type of `tags` be a dict?

Comment (cmanallen) [reply src/sentry/flags/docs/api.md:37]: yes

File: src/sentry/flags/docs/api.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+@cmanallen
+
+**How to read this document.**
+
+This document is structured by resource with each resource having actions that can be performed against it. Every action that either accepts a request or returns a response WILL document the full interchange format. Clients may opt to restrict response data or provide a subset of the request data.
+
+## Flag Logs [/organizations/<organization_id_or_slug>/flag-log/]
+
+- Parameters
+  - query (optional, string) - Search query with space-separated field/value pairs. ie: `?query=environment:prod AND project:3`.
```
Comment (michellewzhang) [inline src/sentry/flags/docs/api.md:16]: we also need ability to pass in `statsPeriod`

File: src/sentry/flags/docs/api.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+- Request (application/json)
+
+  ```json
+  {
+    "data": [
+      {
+        "action": "modified",
+        "flag": "flag-name",
+        "modified_at": "2024-11-19T19:12:55",
+        "modified_by": "colton.allen@sentry.io",
```
Comment (ryan953) [inline src/sentry/flags/docs/api.md:101]: is `modified_by_type` inferred?

**Review by ryan953 — APPROVED** [review]

File: src/sentry/flags/docs/api.md (hunk 8 above)
Comment (cmanallen) [reply src/sentry/flags/docs/api.md:101]: Its inferred from the provider type.  Not from the value of modified_by.  Certain providers give us certain data.  We apply the type based on the provider.

**Pull request merged**